core/tests/test-dependencies/route-to-module-mapping-generator.ts

```diff
@@ -0,0 +1,403 @@
+  return getValueFromLiteralStringOrBinaryExpression(pathProperty);
+};
+
+/**
+ * Gets the module in the route object AST node.
+ */
+const getModuleFromRouteObjectNode = (
+  routeObjectNode: ObjectLiteralExpression
+): string | undefined => {
+  const loadChildrenProperty = routeObjectNode.getProperty('loadChildren');
+  if (loadChildrenProperty === undefined) {
+    return undefined;
+  }
+  const loadChildrenInitializer = loadChildrenProperty
+    .asKindOrThrow(ts.SyntaxKind.PropertyAssignment)
+    .getInitializerOrThrow()
+    .asKindOrThrow(ts.SyntaxKind.ArrowFunction);
+  const loadChildrenCallArgument = loadChildrenInitializer
+    .getBody()
+    .getChildAtIndexIfKindOrThrow(0, ts.SyntaxKind.PropertyAccessExpression)
+    .getChildAtIndexIfKindOrThrow(0, ts.SyntaxKind.CallExpression)
+    .getArguments()[0];
+  return getValueFromLiteralStringOrBinaryExpression(loadChildrenCallArgument);
+};
+
+/**
+ * Gets the path match property in the route object AST node.
+ */
+const getPathMatchFromRouteObjectNode = (
+  routeObjectNode: ObjectLiteralExpression
+): string | undefined => {
+  const pathMatchProperty = routeObjectNode.getProperty('pathMatch');
+  if (pathMatchProperty === undefined) {
+    return undefined;
+  }
+  return pathMatchProperty
+    .asKindOrThrow(ts.SyntaxKind.PropertyAssignment)
+    .getInitializerOrThrow()
+    .asKindOrThrow(ts.SyntaxKind.StringLiteral)
+    .getLiteralText();
+};
+
+/**
+ * Gets the children property in the route object AST node.
+ */
+const getChildrenFromRouteObjectNode = (
+  routeObjectNode: ObjectLiteralExpression
+): ObjectLiteralExpression[] | undefined => {
+  const childrenProperty = routeObjectNode.getProperty('children');
+  if (childrenProperty === undefined) {
+    return undefined;
+  }
+  return childrenProperty
+    .asKindOrThrow(ts.SyntaxKind.PropertyAssignment)
+    .getInitializerOrThrow()
+    .asKindOrThrow(ts.SyntaxKind.ArrayLiteralExpression)
+    .getElements() as ObjectLiteralExpression[];
+};
+
+/**
+ * Converts a route object AST node to a map element.
+ */
+const convertRouteNodeToMap = (
+  routeNode: ObjectLiteralExpression,
+  parentRoutePath?: string,
+  parentRoutingModuleFilePath?: string
+): Map<Route, string> => {
+  let routeToModuleMapping: Map<Route, string> = new Map();
+  const path = getPathFromRouteObjectNode(routeNode);
+  if (path === undefined) {
+    throw new Error('No path was found in the route object.');
+  }
+
+  const module = getModuleFromRouteObjectNode(routeNode);
+  const pathMatch = getPathMatchFromRouteObjectNode(routeNode);
+  const children = getChildrenFromRouteObjectNode(routeNode);
+  const routePath = parentRoutePath ? `${parentRoutePath}/${path}` : path;
+
+  if (children) {
+    for (const child of children) {
+      routeToModuleMapping = extendMap(
+        routeToModuleMapping,
+        convertRouteNodeToMap(
+          child,
+          routePath,
+          parentRoutingModuleFilePath || module
+        )
+      );
+    }
```

Comment:
Can this be in a separate function? Ditto for the ``if (module)` block.

Reply:
Any suggestion? Don't feel like the logic can really be encapsulated, they have different parameters for extendMap.

Reply:
I am not saying integrate them together; to me, it seems a bit more readable to move the whole for loop into a separate function.

Reply:
Done not sure if the changes reflect what you were talking about.